### PR TITLE
Pool or no pool

### PR DESCRIPTION
**Pull request by arlolra**
8dccd56 should have failed but the global agent was never being used because of the default pool settings. There's an incompatibility with v5.7 of node that's fixed in b2c7bb8.

If you want to set that pool then we should set the `agentClass` to this `ConnectTimeoutAgent` presumably.

/cc @wikimedia/services

### CODE REVIEW

**Comment by gwicke** [comment]
@arlolra, I believe #21 has pretty much resolved the same issue. As far as I it, the default agent wasn't used partly because `request` now mangles the agent as well.

I don't see anything left here that we should port, but will let @Pchelolo have a look as well to make sure.

File: index.js [hunk 1]
```diff
@@ -89,11 +117,6 @@ function getOptions(uri, o, method) {
         o.timeout = 2 * 60 * 1000; // 2 minutes
     }
 
-    // Default pool options: Don't limit the number of sockets
-    if (!o.pool) {
-        o.pool = {maxSockets: Infinity};
-    }
-
```
Comment (Pchelolo) [inline index.js:96]: @arlolra The #21 is doing all the same as this one, but it doesn't do this piece. I'm not quite sure why did you remove this? Could you give some context please?

Comment (arlolra) [reply index.js:96]: The commit msg for that change says,

> Setting any pool option will create a new agent of the httpModule
>    class instead of using the global agent. So, the whole rigmarole with
>    the connection timeout was never being used. Not sure what the
>    desired behaviour is here, but it certainly isn't this.

Comment (Pchelolo) [reply index.js:96]: @arlolra oh indeed, sorry to ping you while I could find the answer on my own... We don't use `globalAgent` any more, but instead use `request` `agentClass` feature, so this is not needed any more. I'm closing the PR as all the rest was done in #21

Comment (arlolra) [reply index.js:96]: Ok, sounds good.  Thanks @Pchelolo

Comment (arlolra) [reply index.js:96]: Above I had said,

> If you want to set that pool then we should set the agentClass to this ConnectTimeoutAgent presumably.
